src/modules/discussion/discussion.controller.ts

```
@@ -125,7 +131,36 @@ export class DiscussionController {
   @ApiUnauthorizedResponse({ description: ''})
   @ApiNotFoundResponse({ description: ''})
   @ApiTags('Discussion')
-  updateDiscussionSettings(): string {
+  async updateDiscussionSettings(
+    @Body() setting: SettingsCreateDTO,
+    @Param('discussionId') discussionId: string): Promise<any> {
+
+    const found = await this.discussionModel.findOne({_id: new Types.ObjectId(discussionId)})
+    if (!found){
+      throw new HttpException('Discussion Id does not exist', HttpStatus.NOT_FOUND);
+    }
+ const score = await this.scoreModel.findOne({_id: new Types.ObjectId(setting.score)});
+    if (!score){
+      throw new HttpException('Score Id does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    const calendar = await this.calendarModel.findOne({_id: new Types.ObjectId(setting.calendar)});
+    if (!calendar){
+      throw new HttpException('Calendar Id does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    //Loop through the setting.post_inspiration
+    for await (const post_inspiration of setting.post_inspiration){
+      let found = await this.post_inspirationModel.findOne({_id: [new Types.ObjectId(post_inspiration)]});
+      if(!found){
+        throw new HttpException('Post inspiration Id does not exist', HttpStatus.NOT_FOUND);
+      }
+    }
+    setting.post_inspiration = setting.post_inspiration.concat(setting.post_inspiration);
+    return await this.post_inspirationModel.findOneAndUpdate({_id: discussionId}, setting.post_inspiration, {new: true});
```

This should be updating the discussionSetting Model and should be checking to see if there is setting already there for the discussion. If there isn't one found we should throw an error.